TrackAruco fails with OpenCV 4.7.0

Hi! 
I built OpenCV 4.7 from the sources (with contrib)  but OpenVINS build fails if ENABLE_ARUCO_TAGS is ON:

```
nvidia@ubuntu:~/wks_openVINS$ colcon build --cmake-args -DCMAKE_BUILD_TYPE=Debug --event-handlers console_cohesion+ --packages-select ov_core ov_init ov_msckf ov_eval # ROS2 with verbose output
Starting >>> ov_core 
[Processing: ov_core]                             
[Processing: ov_core]                                     
[Processing: ov_core]                                       
--- output: ov_core                                         
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda-11.4 (found suitable exact version "11.4") 
-- Found OpenCV: /usr/local (found suitable version "4.7.0", minimum required is "4") 
-- Found Boost: /usr/lib/aarch64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found version "1.71.0") found components: system filesystem thread date_time 
-- OPENCV: 4.7.0 | BOOST: 1.71.0
-- PYTHON VERSION: 3.8.10
-- PYTHON INCLUDE: /usr/include/python3.8
-- PYTHON LIBRARIES: /usr/lib/aarch64-linux-gnu/libpython3.8.so
-- Found Python3: /usr/bin/python3.8 (found version "3.8.10") found components: Interpreter 
-- ROS *2* version found, building ROS2.cmake
-- Found rclcpp: 16.0.3 (/home/nvidia/ros2_humble/install/share/rclcpp/cmake)
-- Found rosidl_generator_c: 3.1.4 (/home/nvidia/ros2_humble/install/share/rosidl_generator_c/cmake)
-- Found rosidl_adapter: 3.1.4 (/home/nvidia/ros2_humble/install/share/rosidl_adapter/cmake)
-- Found rosidl_generator_cpp: 3.1.4 (/home/nvidia/ros2_humble/install/share/rosidl_generator_cpp/cmake)
-- Using all available rosidl_typesupport_c: rosidl_typesupport_fastrtps_c;rosidl_typesupport_introspection_c
-- Using all available rosidl_typesupport_cpp: rosidl_typesupport_fastrtps_cpp;rosidl_typesupport_introspection_cpp
-- Found rmw_implementation_cmake: 6.1.1 (/home/nvidia/ros2_humble/install/share/rmw_implementation_cmake/cmake)
-- Found rmw_fastrtps_cpp: 6.2.2 (/home/nvidia/ros2_humble/install/share/rmw_fastrtps_cpp/cmake)
-- Found OpenSSL: /usr/lib/aarch64-linux-gnu/libcrypto.so (found version "1.1.1f")  
-- Found FastRTPS: /home/nvidia/ros2_humble/install/include  
-- Using RMW implementation 'rmw_fastrtps_cpp' as default
-- Found cv_bridge: 3.2.1 (/home/nvidia/ros2_humble/install/share/cv_bridge/cmake)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/nvidia/wks_openVINS/build/ov_core
[ 10%] Building CXX object CMakeFiles/ov_core_lib.dir/src/cpi/CpiV1.cpp.o
[ 10%] Building CXX object CMakeFiles/ov_core_lib.dir/src/dummy.cpp.o
[ 21%] Building CXX object CMakeFiles/ov_core_lib.dir/src/cpi/CpiV2.cpp.o
[ 21%] Building CXX object CMakeFiles/ov_core_lib.dir/src/sim/BsplineSE3.cpp.o
[ 26%] Building CXX object CMakeFiles/ov_core_lib.dir/src/track/TrackAruco.cpp.o
[ 31%] Building CXX object CMakeFiles/ov_core_lib.dir/src/track/TrackBase.cpp.o
[ 36%] Building CXX object CMakeFiles/ov_core_lib.dir/src/track/TrackDescriptor.cpp.o
In file included from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h: In constructor ‘ov_core::TrackAruco::TrackAruco(std::unordered_map<long unsigned int, std::shared_ptr<ov_core::CamBase> >, int, bool, ov_core::TrackBase::HistogramMethod, bool)’:
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:58:77: error: no match for ‘operator=’ (operand types are ‘cv::Ptr<cv::aruco::Dictionary>’ and ‘cv::aruco::Dictionary’)
   58 |     aruco_dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
      |                                                                             ^
In file included from /usr/local/include/opencv4/opencv2/core/cvstd.hpp:81,
                 from /usr/local/include/opencv4/opencv2/core/base.hpp:58,
                 from /usr/local/include/opencv4/opencv2/core.hpp:53,
                 from /usr/local/include/opencv4/opencv2/objdetect/aruco_dictionary.hpp:7,
                 from /usr/local/include/opencv4/opencv2/objdetect/aruco_detector.hpp:7,
                 from /usr/local/include/opencv4/opencv2/aruco.hpp:7,
                 from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:26,
                 from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:121:10: note: candidate: ‘cv::Ptr<T>& cv::Ptr<T>::operator=(const cv::Ptr<T>&) [with T = cv::aruco::Dictionary]’
  121 |     Ptr& operator=(const Ptr& o) { std::shared_ptr<T>::operator =(o); return *this; }
      |          ^~~~~~~~
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:121:31: note:   no known conversion for argument 1 from ‘cv::aruco::Dictionary’ to ‘const cv::Ptr<cv::aruco::Dictionary>&’
  121 |     Ptr& operator=(const Ptr& o) { std::shared_ptr<T>::operator =(o); return *this; }
      |                    ~~~~~~~~~~~^
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:122:38: note: candidate: ‘template<class Y> cv::Ptr<T>& cv::Ptr<T>::operator=(const cv::Ptr<Y>&) [with Y = Y; T = cv::aruco::Dictionary]’
  122 |     template<typename Y> inline Ptr& operator=(const Ptr<Y>& o) { std::shared_ptr<T>::operator =(o); return *this; }
      |                                      ^~~~~~~~
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:122:38: note:   template argument deduction/substitution failed:
In file included from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:58:77: note:   ‘cv::aruco::Dictionary’ is not derived from ‘const cv::Ptr<_Tp>’
   58 |     aruco_dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
      |                                                                             ^
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:59:51: error: ‘create’ is not a member of ‘cv::aruco::DetectorParameters’
   59 |     aruco_params = cv::aruco::DetectorParameters::create();
      |                                                   ^~~~~~
make[2]: *** [CMakeFiles/ov_core_lib.dir/build.make:146: CMakeFiles/ov_core_lib.dir/src/track/TrackAruco.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:141: CMakeFiles/ov_core_lib.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
---
--- stderr: ov_core
In file included from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h: In constructor ‘ov_core::TrackAruco::TrackAruco(std::unordered_map<long unsigned int, std::shared_ptr<ov_core::CamBase> >, int, bool, ov_core::TrackBase::HistogramMethod, bool)’:
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:58:77: error: no match for ‘operator=’ (operand types are ‘cv::Ptr<cv::aruco::Dictionary>’ and ‘cv::aruco::Dictionary’)
   58 |     aruco_dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
      |                                                                             ^
In file included from /usr/local/include/opencv4/opencv2/core/cvstd.hpp:81,
                 from /usr/local/include/opencv4/opencv2/core/base.hpp:58,
                 from /usr/local/include/opencv4/opencv2/core.hpp:53,
                 from /usr/local/include/opencv4/opencv2/objdetect/aruco_dictionary.hpp:7,
                 from /usr/local/include/opencv4/opencv2/objdetect/aruco_detector.hpp:7,
                 from /usr/local/include/opencv4/opencv2/aruco.hpp:7,
                 from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:26,
                 from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:121:10: note: candidate: ‘cv::Ptr<T>& cv::Ptr<T>::operator=(const cv::Ptr<T>&) [with T = cv::aruco::Dictionary]’
  121 |     Ptr& operator=(const Ptr& o) { std::shared_ptr<T>::operator =(o); return *this; }
      |          ^~~~~~~~
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:121:31: note:   no known conversion for argument 1 from ‘cv::aruco::Dictionary’ to ‘const cv::Ptr<cv::aruco::Dictionary>&’
  121 |     Ptr& operator=(const Ptr& o) { std::shared_ptr<T>::operator =(o); return *this; }
      |                    ~~~~~~~~~~~^
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:122:38: note: candidate: ‘template<class Y> cv::Ptr<T>& cv::Ptr<T>::operator=(const cv::Ptr<Y>&) [with Y = Y; T = cv::aruco::Dictionary]’
  122 |     template<typename Y> inline Ptr& operator=(const Ptr<Y>& o) { std::shared_ptr<T>::operator =(o); return *this; }
      |                                      ^~~~~~~~
/usr/local/include/opencv4/opencv2/core/cvstd_wrapper.hpp:122:38: note:   template argument deduction/substitution failed:
In file included from /home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:58:77: note:   ‘cv::aruco::Dictionary’ is not derived from ‘const cv::Ptr<_Tp>’
   58 |     aruco_dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
      |                                                                             ^
/home/nvidia/wks_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:59:51: error: ‘create’ is not a member of ‘cv::aruco::DetectorParameters’
   59 |     aruco_params = cv::aruco::DetectorParameters::create();
      |                                                   ^~~~~~
make[2]: *** [CMakeFiles/ov_core_lib.dir/build.make:146: CMakeFiles/ov_core_lib.dir/src/track/TrackAruco.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:141: CMakeFiles/ov_core_lib.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
---
Failed   <<< ov_core [1min 48s, exited with code 2]

Summary: 0 packages finished [1min 49s]
  1 package failed: ov_core
  1 package had stderr output: ov_core
  3 packages not processed
```

I think it's a problem related to the aruco library that is not in contrib anymore. I think it's in opencv2/objdetect module now.
Adding
```
#include <opencv2/objdetect/aruco_detector.hpp>
#include <opencv2/objdetect/aruco_dictionary.hpp> 
```
in TrackAruco.h doesn't fix the problem.

Hi @ZibYang 
Your fix removed Ptr errors but not this one:

```
Starting >>> ov_core 
Starting >>> ov_data                  
Finished <<< ov_data [1.45s]                                                        
--- stderr: ov_core                              
In file included from /home/nvidia/ws_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
/home/nvidia/ws_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h: In constructor ‘ov_core::TrackAruco::TrackAruco(std::unordered_map<long unsigned int, std::shared_ptr<ov_core::CamBase> >, int, bool, ov_core::TrackBase::HistogramMethod, bool)’:
/home/nvidia/ws_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:59:51: error: ‘create’ is not a member of ‘cv::aruco::DetectorParameters’
   59 |     aruco_params = cv::aruco::DetectorParameters::create();
      |                                                   ^~~~~~
make[2]: *** [CMakeFiles/ov_core_lib.dir/build.make:146: CMakeFiles/ov_core_lib.dir/src/track/TrackAruco.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:141: CMakeFiles/ov_core_lib.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
---
Failed   <<< ov_core [24.7s, exited with code 2]

Summary: 1 package finished [25.9s]
  1 package failed: ov_core
  1 package had stderr output: ov_core
  3 packages not processed
```

> Hi @ZibYang Your fix removed Ptr errors but not this one:
> 
> ```
> Starting >>> ov_core 
> Starting >>> ov_data                  
> Finished <<< ov_data [1.45s]                                                        
> --- stderr: ov_core                              
> In file included from /home/nvidia/ws_openVINS/src/open_vins/ov_core/src/track/TrackAruco.cpp:22:
> /home/nvidia/ws_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h: In constructor ‘ov_core::TrackAruco::TrackAruco(std::unordered_map<long unsigned int, std::shared_ptr<ov_core::CamBase> >, int, bool, ov_core::TrackBase::HistogramMethod, bool)’:
> /home/nvidia/ws_openVINS/src/open_vins/ov_core/src/track/TrackAruco.h:59:51: error: ‘create’ is not a member of ‘cv::aruco::DetectorParameters’
>    59 |     aruco_params = cv::aruco::DetectorParameters::create();
>       |                                                   ^~~~~~
> make[2]: *** [CMakeFiles/ov_core_lib.dir/build.make:146: CMakeFiles/ov_core_lib.dir/src/track/TrackAruco.cpp.o] Error 1
> make[1]: *** [CMakeFiles/Makefile2:141: CMakeFiles/ov_core_lib.dir/all] Error 2
> make: *** [Makefile:146: all] Error 2
> ---
> Failed   <<< ov_core [24.7s, exited with code 2]
> 
> Summary: 1 package finished [25.9s]
>   1 package failed: ov_core
>   1 package had stderr output: ov_core
>   3 packages not processed
> ```

Sorry about that. You should also change  line 59
from  `aruco_params = cv::aruco::DetectorParameters::create();`
to `aruco_params = cv::aruco::DetectorParameters();`
And I believe that error will disappear~ enjoy your day!

ps. My reference is from: https://docs.opencv.org/4.x/d5/dae/tutorial_aruco_detection.html

--- a/.github/workflows/build_ros2.yml
+++ b/.github/workflows/build_ros2.yml
@@ -67,6 +67,6 @@ jobs:
       - name: Run Build in Docker
         run: |
           docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "cd /catkin_ws && colcon build"
-      - name: Run OpenVINS Simulation!
-        run: |
-          docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "cd /catkin_ws && source install/setup.bash && ros2 run ov_msckf run_simulation src/open_vins/config/rpng_sim/estimator_config.yaml"
+#      - name: Run OpenVINS Simulation!
+#        run: |
+#          docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "cd /catkin_ws && source install/setup.bash && ros2 run ov_msckf run_simulation src/open_vins/config/rpng_sim/estimator_config.yaml"

--- a/.github/workflows/build_ros2.yml
+++ b/.github/workflows/build_ros2.yml
@@ -49,3 +49,24 @@ jobs:
       - name: Run OpenVINS Simulation!
         run: |
           docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "cd /catkin_ws && source install/setup.bash && ros2 run ov_msckf run_simulation src/open_vins/config/rpng_sim/estimator_config.yaml"
+  build_2204:
+    name: "ROS2 Ubuntu 22.04"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Code Checkout
+        uses: actions/checkout@v2
+      - name: Create Workspace and Docker Image
+        run: |
+          export REPO=$(basename $GITHUB_REPOSITORY) &&
+          cd $GITHUB_WORKSPACE/.. && mkdir src/ &&
+          mv $REPO/ src/ && mkdir $REPO/ && mv src/ $REPO/ && cd $REPO/ &&
+          docker build -t openvins -f $GITHUB_WORKSPACE/src/$REPO/Dockerfile_ros2_22_04 .
+      - name: Echo Enviroment
+        run: |
+          docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "echo $ROS_DISTRO && echo $ROS_VERSION"
+      - name: Run Build in Docker
+        run: |
+          docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "cd /catkin_ws && colcon build"
+      - name: Run OpenVINS Simulation!
+        run: |
+          docker run -t --mount type=bind,source=$GITHUB_WORKSPACE,target=/catkin_ws openvins /bin/bash -c "cd /catkin_ws && source install/setup.bash && ros2 run ov_msckf run_simulation src/open_vins/config/rpng_sim/estimator_config.yaml"

--- a/Dockerfile_ros2_22_04
+++ b/Dockerfile_ros2_22_04
@@ -1,0 +1,47 @@
+FROM osrf/ros:humble-desktop
+
+# =========================================================
+# =========================================================
+
+# Are you are looking for how to use this docker file?
+#   - https://docs.openvins.com/dev-docker.html
+#   - https://docs.docker.com/get-started/
+#   - http://wiki.ros.org/docker/Tutorials/Docker
+
+# =========================================================
+# =========================================================
+
+# Dependencies we use, catkin tools is very good build system
+# Also some helper utilities for fast in terminal edits (nano etc)
+RUN apt-get update && apt-get install -y libeigen3-dev nano git
+
+# Ceres solver install and setup
+RUN sudo apt-get install -y cmake libgoogle-glog-dev libgflags-dev libatlas-base-dev libeigen3-dev libsuitesparse-dev libceres-dev
+# ENV CERES_VERSION="2.0.0"
+# RUN git clone https://ceres-solver.googlesource.com/ceres-solver && \
+#     cd ceres-solver && \
+#     git checkout tags/${CERES_VERSION} && \
+#     mkdir build && cd build && \
+#     cmake .. && \
+#     make -j$(nproc) install && \
+#     rm -rf ../../ceres-solver
+
+# Seems this has Python 3.10 installed on it...
+RUN apt-get update && apt-get install -y python3-dev python3-matplotlib python3-numpy python3-psutil python3-tk
+
+# Install deps needed for clion remote debugging
+# https://blog.jetbrains.com/clion/2020/01/using-docker-with-clion/
+# RUN sed -i '6i\source "/catkin_ws/install/setup.bash"\' /ros_entrypoint.sh
+RUN apt-get update && apt-get install -y ssh build-essential gcc g++ \
+    gdb clang cmake rsync tar python3 && apt-get clean
+RUN ( \
+    echo 'LogLevel DEBUG2'; \
+    echo 'PermitRootLogin yes'; \
+    echo 'PasswordAuthentication yes'; \
+    echo 'Subsystem sftp /usr/lib/openssh/sftp-server'; \
+  ) > /etc/ssh/sshd_config_test_clion \
+  && mkdir /run/sshd
+RUN useradd -m user && yes password | passwd user
+RUN usermod -s /bin/bash user
+CMD ["/usr/sbin/sshd", "-D", "-e", "-f", "/etc/ssh/sshd_config_test_clion"]
+

--- a/config/euroc_mav/kalibr_imu_chain.yaml
+++ b/config/euroc_mav/kalibr_imu_chain.yaml
@@ -18,7 +18,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/kaist/kalibr_imu_chain.yaml
+++ b/config/kaist/kalibr_imu_chain.yaml
@@ -26,7 +26,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/kaist_vio/kalibr_imu_chain.yaml
+++ b/config/kaist_vio/kalibr_imu_chain.yaml
@@ -22,7 +22,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/rpng_aruco/kalibr_imu_chain.yaml
+++ b/config/rpng_aruco/kalibr_imu_chain.yaml
@@ -18,7 +18,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/rpng_ironsides/kalibr_imu_chain.yaml
+++ b/config/rpng_ironsides/kalibr_imu_chain.yaml
@@ -18,7 +18,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/rpng_plane/kalibr_imu_chain.yaml
+++ b/config/rpng_plane/kalibr_imu_chain.yaml
@@ -18,7 +18,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/rpng_sim/kalibr_imu_chain.yaml
+++ b/config/rpng_sim/kalibr_imu_chain.yaml
@@ -18,7 +18,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/rs_d455/kalibr_imu_chain.yaml
+++ b/config/rs_d455/kalibr_imu_chain.yaml
@@ -27,7 +27,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/rs_t265/kalibr_imu_chain.yaml
+++ b/config/rs_t265/kalibr_imu_chain.yaml
@@ -22,7 +22,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/tum_vi/kalibr_imu_chain.yaml
+++ b/config/tum_vi/kalibr_imu_chain.yaml
@@ -18,7 +18,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/uzhfpv_indoor/kalibr_imu_chain.yaml
+++ b/config/uzhfpv_indoor/kalibr_imu_chain.yaml
@@ -22,7 +22,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/uzhfpv_indoor_45/kalibr_imu_chain.yaml
+++ b/config/uzhfpv_indoor_45/kalibr_imu_chain.yaml
@@ -22,7 +22,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/uzhfpv_outdoor/kalibr_imu_chain.yaml
+++ b/config/uzhfpv_outdoor/kalibr_imu_chain.yaml
@@ -22,7 +22,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/config/uzhfpv_outdoor_45/kalibr_imu_chain.yaml
+++ b/config/uzhfpv_outdoor_45/kalibr_imu_chain.yaml
@@ -22,7 +22,7 @@ imu0:
   model: "kalibr"
   # how to get from Kalibr imu.yaml result file:
   #   - Tw is imu0:gyroscopes:M:
-  #   - R_IMUtoGYRO: is imu0:gyroscopes:M:
+  #   - R_IMUtoGYRO: is imu0:gyroscopes:C_gyro_i:
   #   - Ta is imu0:accelerometers:M:
   #   - R_IMUtoACC not used by Kalibr
   #   - Tg is imu0:gyroscopes:A:

--- a/docs/gs-tutorial.dox
+++ b/docs/gs-tutorial.dox
@@ -79,7 +79,7 @@ For RVIZ, one can open the `ov_msckf/launch/display.rviz` configuration file.
 You should see the system publishing features and a state estimate.
 
 @code{.shell-session}
-rviz # term 2
+rviz -d ov_msckf/launch/display.rviz # term 2
 rosbag play V1_01_easy.bag # term 3
 @endcode
 

--- a/ov_core/src/test_tracking.cpp
+++ b/ov_core/src/test_tracking.cpp
@@ -97,7 +97,7 @@ int main(int argc, char **argv) {
   // Location of the ROS bag we want to read in
   std::string path_to_bag;
   nh->param<std::string>("path_bag", path_to_bag, "/home/patrick/datasets/euroc_mav/V1_01_easy.bag");
-  // nh->param<std::string>("path_bag", path_to_bag, "/home/patrick/datasets/open_vins/aruco_room_01.bag");
+  // nh->param<std::string>("path_bag", path_to_bag, "/home/patrick/datasets/rpng_aruco/aruco_room_01.bag");
   PRINT_INFO("ros bag path is: %s\n", path_to_bag.c_str());
 
   // Get our start location and how much of the bag we want to play

--- a/ov_core/src/track/TrackAruco.cpp
+++ b/ov_core/src/track/TrackAruco.cpp
@@ -94,7 +94,11 @@ void TrackAruco::perform_tracking(double timestamp, const cv::Mat &imgin, size_t
   //===================================================================================
 
   // Perform extraction
+#if CV_MAJOR_VERSION > 4 || ( CV_MAJOR_VERSION == 4 && CV_MINOR_VERSION >= 7)
+  aruco_detector.detectMarkers(img0, corners[cam_id], ids_aruco[cam_id], rejects[cam_id]);
+#else
   cv::aruco::detectMarkers(img0, aruco_dict, corners[cam_id], ids_aruco[cam_id], aruco_params, rejects[cam_id]);
+#endif
   rT2 = boost::posix_time::microsec_clock::local_time();
 
   //===================================================================================

--- a/ov_core/src/track/TrackAruco.h
+++ b/ov_core/src/track/TrackAruco.h
@@ -55,10 +55,16 @@ public:
                       bool downsize)
       : TrackBase(cameras, 0, numaruco, stereo, histmethod), max_tag_id(numaruco), do_downsizing(downsize) {
 #if ENABLE_ARUCO_TAGS
+#if CV_MAJOR_VERSION > 4 || ( CV_MAJOR_VERSION == 4 && CV_MINOR_VERSION >= 7)
+    aruco_dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
+    aruco_params.cornerRefinementMethod = cv::aruco::CORNER_REFINE_SUBPIX;
+    aruco_detector = cv::aruco::ArucoDetector(aruco_dict, aruco_params);
+#else
     aruco_dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
     aruco_params = cv::aruco::DetectorParameters::create();
     // NOTE: people with newer opencv might fail here
     // aruco_params->cornerRefinementMethod = cv::aruco::CornerRefineMethod::CORNER_REFINE_SUBPIX;
+#endif
 #else
     PRINT_ERROR(RED "[ERROR]: you have not compiled with aruco tag support!!!\n" RESET);
     std::exit(EXIT_FAILURE);
@@ -101,11 +107,20 @@ protected:
   bool do_downsizing;
 
 #if ENABLE_ARUCO_TAGS
+#if CV_MAJOR_VERSION > 4 || ( CV_MAJOR_VERSION == 4 && CV_MINOR_VERSION >= 7)
+  // Our dictionary that we will extract aruco tags with
+  cv::aruco::Dictionary aruco_dict;
+  // Parameters the opencv extractor uses
+  cv::aruco::DetectorParameters aruco_params;
+  // Actual detector class
+  cv::aruco::ArucoDetector aruco_detector;
+#else
   // Our dictionary that we will extract aruco tags with
   cv::Ptr<cv::aruco::Dictionary> aruco_dict;
-
   // Parameters the opencv extractor uses
   cv::Ptr<cv::aruco::DetectorParameters> aruco_params;
+#endif
+
 
   // Our tag IDs and corner we will get from the extractor
   std::unordered_map<size_t, std::vector<int>> ids_aruco;

